src/turbo-stream.ts

```diff
@@ -265,7 +300,104 @@ export function encode(
                 }))
             );
           }
-          await Promise.race(Object.values(encoder.deferred));
+          for (const [streamId, stream] of Object.entries(encoder.streams)) {
+            if (stream.finish) continue;
+            const { resolve, promise } = new Deferred<void>();
+            stream.finish = promise;
+            promise.finally(() => {
+              delete encoder.streams[Number(streamId)];
+            });
+            const reader = stream.getReader();
+            consumeChunk();
+            function consumeChunk() {
+              reader
+                .read()
+                .then(({ done, value }) => {
+                  if (signal?.aborted) {
+                    throw signal.reason || new Error("Signal was aborted.");
+                  }
+                  if (done) {
+                    controller.enqueue(
+                      textEncoder.encode(
+                        `${TYPE_STREAM}${TYPE_DONE}${streamId}:[]\n`
```

Comment:
```suggestion
                        `${TYPE_STREAM}${TYPE_DONE}${streamId}\n`
```
This could be modified like this to save a few characters, but it seemed more consistent that way - I'm very open for input there :)